### DISCUSSION TITLE
Feature Request: support parameters in Python `con.sql`

### DISCUSSION THREAD

**Discussion by jcrist** [duckdb/duckdb#8591] [Ideas]
As pointed out in https://github.com/duckdb/duckdb/discussions/8582#discussioncomment-6737792, sometimes using `.sql` may be faster than using `.execute` to execute the same query. For ibis workloads, we'd like to make use of the former when possible. However, currently `.sql` doesn't support passing in parameters, meaning we have to do the parameter substitution client-side.

It would be nice if we could pass in parameters the same ways as done in `con.execute`.

```python
import duckdb

con = duckdb.connect()

# Works today
con.execute("SELECT 1 + ? AS x", parameters=(2,)).arrow()

# Doesn't work, but would be nice if it did
con.sql("SELECT 1 + ? AS x", parameters=(2,)).arrow()
```

**Comment by Tishj** [comment]
```py
import duckdb

con = duckdb.connect()

# Doesn't work, but would be nice if it did
res = con.sql("SELECT 1 + ? AS x", params=(2,)).arrow()
print(res)
```
```
pyarrow.Table
x: int32
----
x: [[3]]
```

**Reply by Tishj** [reply]
Works since 0.9.0 👍
